new_bluesky_instrument.py

```diff
@@ -140,6 +141,8 @@ def branch_name_header(org, repo):
 
 def download_zip(local_zip_file, branch):
     """Download repository as ZIP file."""
+    import requests
+
```

Comment:
Is there any reason why this import is not global?

Reply:
It is not global so the application does not crash under Python <3.9 with this exception.  Instead, the more meaningful message is printed to use Python 3.9 or higher.

Could have put a try..except at the global level...

Reply:
I'll rewrite it to the global part.

Reply:
New commit looks good. If you believe this to be a legitimate worry we will encounter during deployment we can implement this solution so that we have better control over the environment. Especially since the repository comes with an env file. 

https://stackoverflow.com/questions/74857446/how-to-specify-python-version-range-in-environment-yml-file

Reply:
Not sure we need to put a maximum version.  Minimum version for the bluesky environment is specified: https://github.com/BCDA-APS/bluesky_training/blob/7f2b0592c6fef6c7836b9d405d738f2c05dee2d2/bluesky/environments/environment_2024_2.yml#L30

This is different than minimum version to run this installer.  We'll want this soon since the current installer:

- fails for Python 3.6 and 3.7
- does not install the latest environment file